adserver/tasks.py

```diff
@@ -818,6 +865,61 @@ def notify_of_completed_flights():
                 flight
             )
 
+        if flight.auto_renew and not flight.live:
+            # This flight is completed but should be renewed
+            # CONSIDER: Breaking this out into a model method
+            #  although it's hard to re-use this with the
+            #  renewal form since more data is provided there.
+            log.info("Auto-renewing flight %s.", flight)
+
+            new_flight_name = f"{flight.campaign.advertiser} - {timezone.now():%b %Y}"
+            new_flight_slug = slugify(new_flight_name)
+
+            while Flight.objects.filter(slug=new_flight_slug).exists():
+                random_char = get_random_string(1)
+                new_flight_slug = slugify(f"{new_flight_slug}-{random_char}")
+
+            new_flight = Flight(
+                name=new_flight_name,
+                slug=new_flight_slug,
+                start_date=timezone.now().today(),
+                end_date=timezone.now().today() + (flight.end_date - flight.start_date),
+                live=True,
+            )
+
+            # Copy flight fields directly
+            for field in (
+                "targeting_parameters",
+                "priority_multiplier",
+                "traffic_cap",
+                "discount",
+                "cpc",
+                "cpm",
+                "sold_clicks",
+                "sold_impressions",
+                "campaign",
+                "auto_renew",  # New flight will also auto-renew
```

Comment:
This should probably be a constant, so we're keeping this consistent across copying.

Reply:
I don't think it can be easily. Different places are copying different things (some are fully automatic while some have some input). Maybe there's a better way to do it but I think it's not a trivial refactor.